### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode character '\ufffd' in position 944: ordinal not in range(128)

### ISSUE THREAD

**Issue by JumpingSpottedTiger** [jay0lee/got-your-back#125]
Firstly, thank you very much for this great utility.

* I have upgraded to the latest GYB release from https://github.com/jay0lee/got-your-back/releases and I still have this issue.

I can confirm that, and I also can confirm that the latest master also exhibits this behavior.

* I am typing the command as described in the GAM Wiki at https://github.com/jay0lee/got-your-back/wiki

My command is `python gyb.py --email [snip]@gmail.com --action restore-mbox --local-folder [snip] --label-restored "[snip]"` (archlinux, so python 3 is implied).

I am trying to restore an mbox archive of a years-old thunderbird profile. I firstly renamed all the mbox files to include the .mbox extension so gyb would pick up these files.

However, one of my mbox files causes the process to crash with the following error:

```
Traceback (most recent call last):
  File "gyb.py", line 1582, in <module>
    main(sys.argv[1:])
  File "gyb.py", line 1293, in main
    full_message = message.as_bytes()
  File "/usr/lib/python3.6/email/message.py", line 178, in as_bytes
    g.flatten(self, unixfrom=unixfrom)
  File "/usr/lib/python3.6/email/generator.py", line 116, in flatten
    self._write(msg)
  File "/usr/lib/python3.6/email/generator.py", line 181, in _write
    self._dispatch(msg)
  File "/usr/lib/python3.6/email/generator.py", line 214, in _dispatch
    meth(msg)
  File "/usr/lib/python3.6/email/generator.py", line 272, in _handle_multipart
    g.flatten(part, unixfrom=False, linesep=self._NL)
  File "/usr/lib/python3.6/email/generator.py", line 116, in flatten
    self._write(msg)
  File "/usr/lib/python3.6/email/generator.py", line 181, in _write
    self._dispatch(msg)
  File "/usr/lib/python3.6/email/generator.py", line 214, in _dispatch
    meth(msg)
  File "/usr/lib/python3.6/email/generator.py", line 319, in _handle_multipart_signed
    self._handle_multipart(msg)
  File "/usr/lib/python3.6/email/generator.py", line 264, in _handle_multipart
    self.write(subparts)
  File "/usr/lib/python3.6/email/generator.py", line 406, in write
    self._fp.write(s.encode('ascii', 'surrogateescape'))
UnicodeEncodeError: 'ascii' codec can't encode character '\ufffd' in position 944: ordinal not in range(128)
```

**Comment by UnknownGuardian** [comment]
Any luck on this?

**Comment by Maladril** [comment]
Hello,

it seems that GYB doesn't support any non-unicode characters in .mbox emails.

Is there any way to fix that ? It seems like it's just an ascii codec that isn't supporting it.

This makes restores from .mbox veru difficult/impossible if they are originating from any account not in english

**Comment by Maladril** [comment]
I've found a workaround for this.

Run the command --action split on the local folder with the mbox

This apparently only split large .mbox into smaller ones. BUT it will also fix the encoding errors and you will be able to restore the .mbox after the split.

Keep in mind that the split command will leave the original files in the folder and append the splitted one with a number at the end. Remove the originals before restoring to avoid duplicates.

I've also found that mbox restore with GYB is usually a restore without labels. I'm currently trying https://github.com/jay0lee/import-mailbox-to-gmail to better do émbox restores and will keep you posted on how well it works.
